src/taipy/core/data/data_node.py

```diff
@@ -181,10 +181,13 @@ def validity_period(self, val):
     @property  # type: ignore
     @_self_reload(_MANAGER_NAME)
     def expiration_date(self) -> datetime:
-        if not self._last_edit_date:
```

Comment:
Is there any specific reason why we didn't call the getter for `last_edit_date` here before?

Reply:
The expiration_date is already reloading the entity, so by calling _last_edit_date, we are already getting the latest data and calling last_edit_date will only trigger the self reload function again without much use. But now the last_edit_date also checks for the system file modification date and not only the date we're storing in the entity. So we need to call last_edit_date instead of _last_edit_date